### DISCUSSION TITLE
Discussion for Header format

### DISCUSSION THREAD

**Discussion by KoukiHama** [eclipse-sw360/sw360-frontend#22] [General]
We should follow sw360 headers or consider to use new one?

https://github.com/eclipse-sw360/sw360-frontend/blob/main/CONTRIBUTING.md#source-license-headers-and-copyrights

（this discussion from [Slack](https://sw360chat.slack.com/archives/C68C2RYS3/p1683169777295319?thread_ts=1683112033.926759&cid=C68C2RYS3) ）

**Comment by KoukiHama** [comment]
It is not good to have different headers for sw360 and sw360 frontend as well as Code of conduct.

However, if it is also different from the Eclipse Foundation's Suggestion, it is worth considering aligning both with it.

My idea is let's use the same header format as SW360 for now.

On the other hand, let's continue the discussion on whether to make the headers for both sw360 and sw360frontend the same as Eclipse Foundation.



https://www.eclipse.org/projects/handbook/#ip-copyright-headers

**Comment by heliocastro** [comment]
Eclipse foundation says three possibilities:
1 - The current one that exists in SW360
2 - The second one that was the one i added as alternative
OR 3- A mixed solution

I decided for number 2 exactly for the same reasons are stated there:
Quoting:
`Alternatively, project teams can adopt a more generic copyright header for each file and include the identities of the copyright holders in the notice file.`

Reference: https://www.eclipse.org/projects/handbook/#ip-copyright-headers

So, in no way whatsoever I'm removing the Companies or Authors, the direct reason was to simplify work for scanners to extract copyrights.

Pure example what is happening today, in some of the usual scanners and not good attending compliance developers after extract in a project NOTICE file:

```plaintext
Copyright 2002 Someone-1
Copyright 2002 Somoone-2
Copyright 2002 Company-1
Copyright 2002-2007 Company-1
Copyright 2002 Company-1
```

And goes on and on, because the inability of some scanners to collect all copyrights from all files, separate and properly avoid deduplications, etc..
Technically, from the point of view of scanners they even need to parse everything, so create a end result more complicated thsan usual.

And i in personal believe think that this is clean and the code will be way more easy to parse.
@ag4ums Is there any reason that you really want to keep the format beyond technical please do so, but just to say, we are compliant with Eclipse Foundation, so from that part is nothing wrong.

We can change to add the mention of NOTICE file a little to looks like the option 2 ( adding extra lines ), but the Copyrights is really not make difference except you think that someone or some AI will copy part of the code and then the copyrights will be lost, which would be valid in both cases.

So, i'm ok to restore the header for old way, need to change the parser check if you  too vote in favor, but again:

- There's nothing wrong related to Eclipse project recommendations, is even part of their recommendations
- Technically is bette. IF 10 different entities touch a file, you end up with a lost of 10 entries per file, which is really hard to processed

**Reply by ag4ums** [reply]
Hello @heliocastro and @KoukiHama
> And goes on and on, because the inability of some scanners to collect all copyrights from all files, separate and properly avoid deduplications, etc..
Technically, from the point of view of scanners they even need to parse everything, so create a end result more complicated thsan usual.
1. Today's scanners like fossology and scancode good to detect all the copyright, some false positive copyright/s may appear but those are eliminated with some rules or ml approaches. Scanners has to go through all files, can not depend on a single file until its a uniform standard.
> Technically is bette. IF 10 different entities touch a file, you end up with a lost of 10 entries per file, which is really hard to processed
2.  Small changes in code base cannot claim copyrights, there has to be significant change to claim copyright in a file, so lots of copyright in a file is very rarely possible.
> We can change to add the mention of NOTICE file a little to looks like the option 2 ( adding extra lines ), but the Copyrights is really not make difference except you think that someone or some AI will copy part of the code and then the copyrights will be lost, which would be valid in both cases.
3. If files are used by other projects the copyright holder(s) will most probably not copied, with a copyrights moved to other file(Notice file) its easy to overlook,
According to the eclipse rulebook and resue conformant. It's Ok to do it, but its an alternative way,
 
To avoid all these confusion I think its always good to have the Copyright Notice in the files. and personally will prefer the old way with resue compliant.

**Reply by heliocastro** [reply]
Ok, i will do to @KoukiHama to mediate this.

If someone wants to copy code, they will do with or without copyrights and if they want to alter they will do it. Only a snippet scan can catch this things.

So, let's Kouki decide, because both formats are accepted, is just question of taste.

**Reply by KoukiHama** [reply]
>  So, let's Kouki decide, because both formats are accepted, is just question of taste.

Let's restore the header for old way and apply the original SW360 copyright notice to SW360-frontend as it is now. (following @ag4ums 's idea)

There are advantages and disadvantages to both methods of copyright notation and it i
s  just question of taste...

I consider it the worst situation to use different methods for SW360 and SW360frontend.
Changing SW360 to current SW360frontend's style takes a lot of work, but the reverse is easy now.

In the meantime, this discussion needs to continue, including what to do about copyrights in the files of SW360 licenses that have not been renewed for four years.

https://github.com/eclipse-sw360/sw360/blob/91ee5d593a61d1e1327d75ad361030a326f9524f/LICENSE#L387

Unfortunately, I could not find any old discussions.　We should clarify when and how we will update this copyright.

**Comment by KoukiHama** [comment]
Note : https://www.eclipse.org/sw360/docs/development/dev-dod-and-style/#licensing-and-file-header

They are sentences about headers in sw360 website

**Reply by ag4ums** [reply]
> Note : https://www.eclipse.org/sw360/docs/development/dev-dod-and-style/#licensing-and-file-header
> 
> They are sentences about headers in sw360 website

Not resue compliant,

**Comment by heliocastro** [comment]
Btw, is reuse compliant, both ones.
We do have a copyright line. Is just not have the names of everyone.

**Comment by heliocastro** [comment]
Headers reverted.
